### PR TITLE
Add OlympicsReplayIE.

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Closes https://github.com/yt-dlp/yt-dlp/issues/897

### CODE REVIEW

File: yt_dlp/extractor/olympics.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import unified_strdate
+
+
+class OlympicsReplayIE(InfoExtractor):
+    _VALID_URL = r'(?:https?://)(?:www\.)?olympics\.com/tokyo-2020/(?:[a-z]{2}/)?replay/(?P<id>[^/#&?]+)'
```
Comment (pukkandan) [inline yt_dlp/extractor/olympics.py:9]:
```suggestion
    _VALID_URL = r'(?:https?://)(?:www\.)?olympics\.com/[^/]+]/(?:[a-z]{2}/)?replay/(?P<id>[^/#&?]+)'
```

File: yt_dlp/extractor/olympics.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }, {
+        'url': 'https://olympics.com/tokyo-2020/en/replay/bd242924-4b22-49a5-a846-f1d4c809250d/mens-bronze-medal-match-hun-esp',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        id = self._match_id(url)
+        # The parameters are hardcoded in the webpage, it's not necessary to download the webpage just for these parameters.
+        # If in downloading webpage serves other functions aswell, then extract these parameters from it.
+        token_url = 'https://appovptok.ovpobs.tv/api/identity/app/token?api_key=OTk5NDcxOjpvY3N3LWFwaXVzZXI%3D&api_secret=ODY4ODM2MjE3ODMwYmVjNTAxMWZlMDJiMTYxZmY0MjFiMjMwMjllMjJmNDA1YWRiYzA5ODcxYTZjZTljZDkxOTo6NTM2NWIzNjRlMTM1ZmI2YWNjNmYzMGMzOGM3NzZhZTY%3D'
```
Comment (pukkandan) [inline yt_dlp/extractor/olympics.py:31]: This is generally not a robust approach, but given the nature of the website, I guess it's fair enough

File: yt_dlp/extractor/olympics.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        token = self._download_webpage(token_url, id)
+        headers = {'x-obs-app-token': token}
+        data_json = self._download_json(f'https://appocswtok.ovpobs.tv/api/schedule-sessions/{id}?include=stream',
+                                        id, headers=headers)
+        meta_data = data_json['data']['attributes']
+        for t_dict in data_json['included']:
+            if t_dict.get('type') == 'Stream':
+                stream_data = t_dict['attributes']
+        m3u8_url = self._download_json(f'https://meteringtok.ovpobs.tv/api/playback-sessions?alias={stream_data["alias"]}&stream={stream_data["stream"]}&type=vod',
+                                       id, headers=headers)['data']['attributes']['url']
```
Comment (pukkandan) [inline yt_dlp/extractor/olympics.py:41]:
```suggestion
        m3u8_url = self._download_json(
            'https://meteringtok.ovpobs.tv/api/playback-sessions', id, headers=headers, query={
                'alias': stream_data['alias'],
                'stream': stream_data['stream'],
                'type': 'vod'
            })['data']['attributes']['url']
```

File: yt_dlp/extractor/olympics.py (hunk 1 above)
Comment (Ashish0804) [reply yt_dlp/extractor/olympics.py:9]: I don't think the site offers older olympics replays 🤔 
And for future ones the url pattern is diff.
Next one is Beijing 2022, imo it's better for it to break and we revisit it in 2022 cuz the site already uses diff way to get videos from the same  tokyo 2020

**Pull request merged**